### PR TITLE
[cmake] add option for `OPENTHREAD_CONFIG_NET_DIAG_VENDOR_INFO_SET_API_ENABLE`

### PR DESCRIPTION
**Pull request by suveshpratapa**
This makes it easier to configure this option for testing on OTBR/RCP.

### CODE REVIEW

File: etc/cmake/options.cmake [hunk 1]
```diff
@@ -321,6 +321,7 @@ ot_multi_option(OT_LOG_OUTPUT OT_LOG_OUTPUT_VALUES OPENTHREAD_CONFIG_LOG_OUTPUT
 
 # - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 
+ot_string_option(OT_NETDIAG_VENDOR_INFO OPENTHREAD_CONFIG_NET_DIAG_VENDOR_INFO_SET_API_ENABLE "Allow setting vendor info at runtime")
```
Comment (abtink) [inline etc/cmake/options.cmake:324]: This should not be a "string option". It is a normal ON/OFF option, so should use `ot_option()` for it.
Also suggest moving this within the list of `ot_option()` definitions above in this file (the list is sorted).

**Comment by abtink** [comment]
> cc @jwhui Looking to enable this for reference builds in the test harness.

Thanks @suveshpratapa.

If the goal is to enable this for reference devices using a CMake flag, an alternative approach could be to change the default value for this config in `src/config/network_diagnostic.h` as follows:

```c++
#ifndef OPENTHREAD_CONFIG_NET_DIAG_VENDOR_INFO_SET_API_ENABLE
#define OPENTHREAD_CONFIG_NET_DIAG_VENDOR_INFO_SET_API_ENABLE OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE
#endif
```

This way, it's enabled by default when `OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE` is used. 

I feel this would be simpler and would apply for any other reference device builds.

Thoughts?

**Comment by abtink** [comment]
> That change sounds fine by me. Are you opposed to also exposing it as a cmake build option?

No. Both are fine for sure.

**Review by abtink — APPROVED** [review]
👍

**Review by jwhui — APPROVED** [review]

**Pull request merged**